pkg/image/controller/trigger/trigger_test.go

```diff
@@ -0,0 +1,1036 @@
+			Informer: podInformer,
+			TriggerFn: func(prefix string) trigger.Indexer {
+				return annotations.NewAnnotationTriggerIndexer(prefix)
+			},
+			Reactor: podReactor,
+		},
+	)
+	c.resourceFailureDelayFn = func(_ int) (time.Duration, bool) {
+		return 0, true
+	}
+	isFn := c.syncImageStreamFn
+	c.syncImageStreamFn = func(key string) error {
+		if err := isFn(key); err != nil {
+			if kapierrs.IsConflict(err) {
+				return err
+			}
+			t.Fatalf("failure on %s: %v", key, err)
+		}
+		return nil
+	}
+	resFn := c.syncResourceFn
+	c.syncResourceFn = func(key string) error {
+		if err := resFn(key); err != nil {
+			if kapierrs.IsConflict(err) {
+				glog.V(4).Infof("conflict syncing resource %s: %v", key, err)
+				return err
+			}
+			t.Fatalf("failure on %s: %v", key, err)
+		}
+		return nil
+	}
+	go isInformer.Run(stopCh)
+	go bcInformer.Run(stopCh)
+	go podInformer.Run(stopCh)
+	go dcInformer.Run(stopCh)
+	go c.Run(8, stopCh)
+
+	numReferencedStreams := int32(float32(numStreams) * ratioReferencedStreams)
+
+	// generate an initial state
+	for i := int32(0); i < numBuildConfigs; i++ {
+		if i < int32(float32(numBuildConfigs)*ratioTriggeredBuildConfigs) {
+			// builds that point to triggers
+			if err := bcWatch.Add(benchmark_1_buildConfig(rnd, i, numReferencedStreams, numTagsPerStream, maxTriggersPerBuild)); err != nil {
+				t.Fatal(err)
+			}
+		} else {
+			// builds that have no image stream triggers
+			if err := bcWatch.Add(benchmark_1_buildConfig(rnd, i, numStreams, numTagsPerStream, 0)); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	for i := int32(0); i < numPods; i++ {
+		// set initial pods
+		podWatch.Add(benchmark_1_pod(rnd, i, numReferencedStreams, numTagsPerStream, maxContainersPerPod))
+	}
+	for i := int32(0); i < numDeploymentConfigs; i++ {
+		// set initial deployments
+		if err := dcWatch.Add(benchmark_1_deploymentConfig(rnd, i, numReferencedStreams, numTagsPerStream, maxContainersPerPod)); err != nil {
+			t.Fatal(err)
+		}
+	}
+	for i := int32(0); i < numStreams; i++ {
+		// set initial image streams
+		if err := isWatch.Add(benchmark_1_imageStream(i, numTagsPerStream, 1, 0, 0)); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	describe := map[string][]string{}
+
+	// make a set of modifications to the streams or builds, verifying after each round
+	for round := 1; round <= rounds; round++ {
+		var changes []interface{}
+		for i := 0; i < iterations; i++ {
+			switch f := rnd.Float32(); {
+			case f < ratioStreamChanges:
+				streamNum := rnd.Int31n(numStreams)
+				stream := benchmark_1_imageStream(streamNum, numTagsPerStream, int32(2+(round-1)*500+i), round, i)
+				existing, _, _ := isInformer.GetStore().GetByKey(stream.Namespace + "/" + stream.Name)
+				stream.ResourceVersion = existing.(*imageapi.ImageStream).ResourceVersion
+				if err := isWatch.Modify(stream); err != nil {
+					glog.V(1).Infof("[round=%d change=%d] failed to modify image stream: %v", round, i, err)
```

Comment:
Why not t.Logf ? Is anybody going to run tests with loglevel configured if they are not familiar with the test?